test/e2e/auth/audit.go

```diff
@@ -23,83 +23,623 @@ import (
+					framework.ExpectNoError(err, "failed to patch secret")
+
+					_, err = f.ClientSet.Core().Secrets(f.Namespace.Name).List(metav1.ListOptions{})
+					framework.ExpectNoError(err, "failed to list secrets")
+
+					err = f.ClientSet.Core().Secrets(f.Namespace.Name).Delete(secret.Name, &metav1.DeleteOptions{})
+					framework.ExpectNoError(err, "failed to delete audit-secret")
+
+				},
+				[]auditEvent{
+					{
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets", namespace),
+						"create",
+						201,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets/audit-secret", namespace),
+						"get",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets", namespace),
+						"list",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseStarted,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+						"watch",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+						"watch",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets/audit-secret", namespace),
+						"update",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets/audit-secret", namespace),
+						"patch",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					}, {
+						v1beta1.LevelMetadata,
+						v1beta1.StageResponseComplete,
+						fmt.Sprintf("/api/v1/namespaces/%s/secrets/audit-secret", namespace),
+						"delete",
+						200,
+						auditTestUser,
+						"secrets",
+						namespace,
+						false,
+						false,
+					},
+				},
+			}, {
+				"CustomResourceDefinition",
+				func() {
+					_, err = testserver.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.ClientPool)
+					framework.ExpectNoError(err, "failed to create custom resource definition")
+					testserver.DeleteCustomResourceDefinition(crd, apiExtensionClient)
```

Comment:
Is CRD (object, not definition) being created here?

Reply:
If you are talking if this command sends actual request to the apiserver then yes.